src/index.ts

```diff
@@ -240,43 +272,89 @@ export type startBlock = () => bigint;
  * }
  */
 export async function* queryLive<T = DefaultType>(
-  userRequest: Request<T> & {
+  userRequest: IS.Request<T> & {
     abortSignal?: AbortSignal;
     startBlock?: startBlock;
+    retryConfig?: Partial<RetryConfig>;
+    timeout?: number;
   },
-): AsyncGenerator<Response<T>> {
+): AsyncGenerator<IS.Response<T>, void, unknown> {
   let running = true;
-  const signals: AbortSignal[] = [];
-  signals.push(AbortSignal.timeout(60_000));
-  if (userRequest.abortSignal) {
-    userRequest.abortSignal.addEventListener("abort", () => {
-      running = false;
-    });
-    signals.push(userRequest.abortSignal);
-  }
-  do {
+  let attempt = 0;
+  const config = {
+    ...DEFAULT_RETRY_CONFIG,
+    timeout: userRequest.timeout ?? DEFAULT_TIMEOUT,
+    ...userRequest.retryConfig,
+    get delay(): number {
+      return Math.min(this.baseDelay * Math.pow(2, attempt - 1), this.maxDelay);
+    },
+  };
+
+  userRequest.abortSignal?.addEventListener("abort", () => {
+    running = false;
+  });
+
+  while (running) {
+    const timeoutSignal = AbortSignal.timeout(config.timeout);
+    const signals = [timeoutSignal];
+
+    if (userRequest.abortSignal) {
+      signals.push(userRequest.abortSignal);
+    }
+
     try {
       const response = await fetch(url("query-live", userRequest), {
         signal: AbortSignal.any(signals),
       });
+
       if (response.status !== 200) {
-        throw new Error(`Index Supply API error: ${response.status}`);
+        if (response.status === 429) {
+          throw `Rate limited, retrying in ${config.delay}ms`;
+        } else if (response.status === 408) {
+          debug("Timeout error, retrying...");
+          // retry immediately
+          continue;
```

Comment:
I merged this block back in from a function, it makes the logic much more straightforward and we can just call `continue` for 408s, and throw for other errors which will hit the `catch` block and delay appropriately.